sparksteps/pricing.py

```diff
@@ -128,7 +150,19 @@ def get_bid_price(ec2_client, pricing_client, instance_type):
     """
     history = get_spot_price_history(ec2_client, instance_type, SPOT_PRICE_LOOKBACK)
     by_zone = price_by_zone(history)
-    zone_profile = get_zone_profile(by_zone)
+    if availability_zone:
+        # Consider only the AZ in which we expect to launch instances.
+        zone_profile = get_zone_profile({availability_zone: by_zone.get(availability_zone, [])})
```

Comment:
`get_zone_profile` will raise exceptions if an empty list is provided as data for an AZ. It'd probably be better to directly fall back to demand price when `availability_zone` is specified but not found in `by_zone`.